### PR TITLE
Added attributes and erase_kdl attribute

### PR DESCRIPTION
**Pull request by algebraic-dev**

### CODE REVIEW

File: test.kind2 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#erase_kdl
```
Comment (nicolas-abril) [inline test.kind2]: I think you forgot to update to kdl_erase

File: tests/suite/to_kdl/erased_2.kind2 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Ata #A : U60
+Ata = 2
+
+#kdl_erase
+Be #B : U60
+Be = 3
```
Comment (nicolas-abril) [inline tests/suite/to_kdl/erased_2.kind2:6]: It'd be good to test that this function can still be callable from other places. Eg:
```
FnA #A {
 (+ FnB 1)
}

#kdl_erase
FnB #B {
 2
}
```
Should generate
```
fun (A) {
 (+ B 1)
}
```
Of course this'll only run in kindelia for U120 operations or native IO functions, but it's a nice test

File: src/book.rs [hunk 3]
```diff
@@ -17,6 +17,13 @@ use crate::book::term::Term;
 use std::collections::HashMap;
 use std::fmt::{Display, Error, Formatter};
 
+#[derive(Clone, Debug)]
+pub struct Attribute {
+    pub name: Ident,
+    pub value: Option<Ident>,
```
Comment (nicolas-abril) [inline src/book.rs:23]: Why only accept a single identifier?
Wouldn't it be better to accept any string and then the attribute handlers parse them?
I can imagine there being future attributes which accept other kinds of things, like numbers, multiple identifiers or even some kind terms.

File: src/book.rs [hunk 4]
```diff
@@ -17,6 +17,13 @@ use crate::book::term::Term;
 use std::collections::HashMap;
 use std::fmt::{Display, Error, Formatter};
 
+#[derive(Clone, Debug)]
+pub struct Attribute {
```
Comment (nicolas-abril) [inline src/book.rs:21]: Why not use an enum with possible attributes here? Not that I don't like this, just want to know your reasoning

File: tests/suite/to_kdl/run.kind2 [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Ata #A : U60
+Ata = 2
+
+#kdl_run
```
Comment (nicolas-abril) [inline tests/suite/to_kdl/run.kind2:4]: Put these functions that become run blocks in a different order to test that run goes last after compilation (eg. Be -> Ata -> Ce here, or change what has the attribute)

**Review by nicolas-abril — CHANGES_REQUESTED** [review]

File: src/book.rs (hunk 3 above)
Comment (algebraic-dev) [reply src/book.rs:23]: At least for now, we are not using values for attributes. I think we should update it when attributes that need this feature come up.

File: src/book.rs (hunk 4 above)
Comment (algebraic-dev) [reply src/book.rs:21]: I don't want to add complexity to the parsing step.

File: test.kind2 (hunk 1 above)
Comment (algebraic-dev) [reply test.kind2]: This test is in the wrong place.

File: tests/suite/to_kdl/erased_2.kind2 (hunk 2 above)
Comment (algebraic-dev) [reply tests/suite/to_kdl/erased_2.kind2:6]: Added.

**Review by nicolas-abril — APPROVED** [review]

**Pull request merged**